NBISweden/vector-oligo-search
# Feat/add tag sequence lookup

This PR closes #17

## It will
- Add new search option lookup type
- Add lookup type "Tag" and rename previous implementation to "KO"
- Add reference script for Tag lookups and tests to verify that we produce the same result
- Make filenames of downloads reflect which type of lookup was used
- Move resource loading for searches into cached function instead of global scope

## To test
- `docker compose up`
- select lookup "Tag"
- Use any of the gene-ids from https://github.com/NBISweden/vector-oligo-search/blob/feat/add-tag-sequence-lookup/search/tests/tag_gene_ids.py
- Submit
- Expect a successful result
- `docker compose run --rm web pytest ./search`
- Expect all tests to succeed
